### PR TITLE
Update typed property array => ArrayCollection to prevent errors

### PR DESCRIPTION
**Pull request by ghost**
When using strict type checking you could get errors like:
```
TypeError: Cannot assign Doctrine\Common\Collections\ArrayCollection to property App\Entity\Category::$products of type array
```

### CODE REVIEW

File: doctrine/associations.rst [hunk 1]
```diff
@@ -220,7 +220,7 @@ class that will hold these objects:
             // ...
 
             #[ORM\OneToMany(targetEntity: Product::class, mappedBy: 'category')]
-            private array $products;
+            private ArrayCollection $products;
```
Comment (xabbuh) [inline doctrine/associations.rst]:
```suggestion
            private Collection $products;
```

Comment (xabbuh) [reply doctrine/associations.rst]: because when loaded from the database the property will be populated with a `PersistentCollection` by Doctrine

Comment (ghost) [reply doctrine/associations.rst]: Ah yes and that implements `Collection` interface agreed.

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Good catch, thanks @johanadivare.

**Pull request merged**